src/resolver.ts

```diff
@@ -1199,14 +1223,42 @@ export class Resolver extends DiagnosticEmitter {
     /** How to proceed with eventualy diagnostics. */
     reportMode: ReportMode = ReportMode.REPORT
   ): Element | null {
-    // TODO
-    if (reportMode == ReportMode.REPORT) {
-      this.error(
-        DiagnosticCode.Operation_not_supported,
-        expression.range
-      );
+    let left = expression.left;
+    let right = expression.right;
+    let leftElement = this.resolveExpression(left, flow, contextualType, reportMode);
+    if (!leftElement) return null;
+    let leftType = (<TypedElement>leftElement!).type;
```

Comment:
If it's always a `TypedElement`, I think `resolveExpression` should have it as the return type. Unless it's not always one, ofc, which would mean it misses something here.

Reply:
That's what I thought, but what's an example of something that does not resolve to a `typedElement`?

Reply:
What can happen there for example is that if an expression referencing a class name is resolved, the element is a static `ClassPrototype` without a type. The same should be true for function names, because the function can be generic with multiple instances so a `FunctionPrototype` is returned. There might be others that I forgot.

Reply:
I see. Is it possible to circumvent this by using `resolveType` here instead of `resolveExpression`?

Reply:
`resolveType` is meant for type nodes exclusively here, so I think it isn't applicable. If I'm not mistaken the resolver should actually return the `ClassPrototype` respectively `FunctionProtoype` here (since we want the element), and whatever the caller of `resolveExpression` is would deal with it appropriately.